### DISCUSSION TITLE
👋👋🌈🎆🥳🎉 Welcome to Bloom Discussions!👋👋🌈🎆🥳🎉

### DISCUSSION THREAD

**Discussion by navnavnav** [bloombloombloom/Bloom#20] [Announcements]
## 👋 Welcome!

Hi all,
 
I've setup GitHub Discussions for Bloom users to ask questions and share anything Bloom related.
Feel free to start a new discussion.

Thanks,
Nav

**Comment by cbwolfe94** [comment]
![Screenshot from 2022-03-18 08-40-48](https://user-images.githubusercontent.com/37260117/159004722-bb45150f-692a-4299-96fd-c5473f3b0cbc.png)
I've downloaded all of the necessary software. Verified my connection to my atmega328p by flashing a program to it using the Atmel ICE using the SPI interface. Not sure what I am doing wrong.

**Reply by navnavnav** [reply]
> I've downloaded all of the necessary software. Verified my connection to my atmega328p by flashing a program to it using the Atmel ICE using the SPI interface. Not sure what I am doing wrong.

Hi @cbwolfe94 

It looks like the target's DWEN fuse bit is not enabled. Bloom can enable this for you - make sure the target is connected to the debug tool via the SPI connection, and then set the `updateDwenFuseBit` to `true` in your target config (in bloom.json).

```
"target": {
    "name": "atmega328p",
    "physicalInterface": "debug-wire",
    "updateDwenFuseBit": true
}
```

See https://bloom.oscillate.io/docs/debugging-avr-debugwire for more on the DWEN fuse bit.

Thanks

**Reply by cbwolfe94** [reply]
Was missing the last line. Will try right now.

**Reply by navnavnav** [reply]
You may have to cycle the target power, just once, after the fuse bit is updated.

**Reply by cbwolfe94** [reply]
Working. Thanks!

**Reply by cbwolfe94** [reply]
Power cycle was indeed necessary.

**Reply by navnavnav** [reply]
Just one more thing to note with debugging via debugWire: If you're using AVRDUDE to program the target, you may run into issues when the DWEN fuse is set, where AVRDUDE will report errors such as "Unexpected response" or "Short read, read only 0 out of X bytes". And it will leave the Atmel-ICE in a bad state, where it will need to be re-enumerated before you can use it again. This is due to a timeout bug in AVRDUDE, explained here: https://github.com/avrdudes/avrdude/issues/900 The PR for this fix is still pending approval.

If you do run into that issue, you can use the `disableDebugWirePreDisconnect` target config param in Bloom, to have Bloom disable the DWEN fuse just before disconnecting with the target. But the down-side to this is that you'll have to cycle the target power every time you start/stop Bloom, or for every debug session, if `releasePostDebugSession` is enabled. So it may be a good idea to disable `releasePostDebugSession` (debug tool config param - see https://bloom.oscillate.io/docs/configuration#debug-tool-config-param-releasePostDebugSession), so that Bloom doesn't disconnect from the target at the end of each debug session.

For more on this, see https://bloom.oscillate.io/docs/debugging-avr-debugwire

**Reply by cbwolfe94** [reply]
Noted. Also I like your profile pic. I am a Chimchar man myself, but I do like Turtwig.

**Reply by navnavnav** [reply]
Thanks! 😊
